spec.bs

```diff
@@ -491,12 +494,243 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component<
 
 A {{URLPattern}} is associated with multiple <dfn>component</dfn> [=structs=].
 
-A [=component=] has an associated <dfn for=component>pattern string</dfn>, a string, initially null.
+A [=component=] has an associated <dfn for=component>pattern string</dfn>, a [=/pattern string=], initially null.
 
 A [=component=] has an associated <dfn for=component>regular expression</dfn>, a {{RegExp}}, initially null.
 
 A [=component=] has an associated <dfn for=component>group name list</dfn>, an array of strings, initially an empty array.
 
+<h2 id=patterns>Patterns</h2>
+
+A <dfn>pattern string</dfn> is a string conforming to a particular syntax.  This pattern syntax is directly based on the syntax used by the popular [path-to-regexp](https://github.com/pillarjs/path-to-regexp) JavaScript library.
+
+<h3 id=parsing-patterns>Parsing Patterns</h3>
+
+<h4 id=tokens>Tokens</h4>
+
+A <dfn>token list</dfn> is a [=list=] containing zero or more [=token=] [=structs=].
+
+A <dfn>token</dfn> is a [=struct=] representing a single lexical token within a [=/pattern string=].
+
+A [=token=] has an associated <dfn for=token>type</dfn>, a string, initially "[=token/type/invalid-char=]".  It must be one of the following:
```

Comment:
Linking to enum values while getting them to be in code font is annoying in Bikeshed. The best I've found is `"<a for="token/type"><code>invalid-char</code></a>"`. I'm not sure it's worth it...

Reply:
Switched them over using backticks instead of `<code>`.